server/src/main/java/com/cloud/usage/UsageServiceImpl.java

```diff
@@ -328,9 +333,20 @@ public boolean generateUsageRecords(GenerateUsageRecordsCmd cmd) {
                     break;
                 case UsageTypes.IP_ADDRESS:
                     IPAddressVO ip = _ipDao.findByUuidIncludingRemoved(usageId);
-                    if (ip != null) {
-                        usageDbId = ip.getId();
+                    if (ip == null) {
+                        break;
+                    }
+                    Long networkId = ip.getAssociatedWithNetworkId();
+                    if (networkId == null) {
+                        networkId = ip.getSourceNetworkId();
+                    }
+                    NetworkDetailVO networkDetail = _networkDetailsDao.findDetail(networkId, Network.hideIpAddressUsage);
+                    if (networkDetail != null && networkDetail.getValue() != null && networkDetail.getValue().equals("true")) {
```

Comment:
As this code is in a test `if (usageId != null)`, it will remove IP address usage only if the API command `listUsageRecords` was called with a specific `usageId` parameter.
For API calls like `command=listUsageRecords&startdate=xxx&enddate=xxx`, it will include all usage records, including for IP addresses on networks with "Hide usage".